### PR TITLE
[modernize] use-default-member-init

### PR DESCRIPTION
**Pull request by Rechi**
## Description
run clang-tidy `modernize-use-default-member-init` check and applied the suggested fixes
run clang-tidy-diff with `modernize-use-nullptr` check and applied the suggested fixes
run clang-tidy `modernize-use-default-member-init` check and applied the suggested fixes
run clang-tidy-diff with `modernize-use-equals-default` check and applied the suggested fixes

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: tools/depends/native/TexturePacker/src/decoder/GifHelper.cpp [hunk 1]
```diff
@@ -43,14 +43,7 @@ int ReadFromVfs(GifFileType* gif, GifByteType* gifbyte, int len)
...
 
-GifHelper::GifHelper() :
-  m_imageSize(0),
-  m_pitch(0),
-  m_loops(0),
-  m_numFrames(0),
-  m_filename(""),
-  m_gif(nullptr),
-  m_pTemplate(nullptr)
+GifHelper::GifHelper() : m_filename("")
```
Comment (ksooo) [inline tools/depends/native/TexturePacker/src/decoder/GifHelper.cpp]: `m_filename`is a `std::string` and its default ctor contructs an empty string. So, no need to call the ctor with `""`here.

File: tools/depends/native/TexturePacker/src/decoder/GifHelper.h [hunk 2]
```diff
@@ -101,15 +101,15 @@ class GifHelper
...
+  unsigned int m_pitch = 0;
+  unsigned int m_loops = 0;
+  unsigned int m_numFrames = 0;
 
   std::string     m_filename;
-  GifFileType*    m_gif;
+  GifFileType* m_gif = nullptr;
   std::vector<GifColor> m_globalPalette;
-  unsigned char*  m_pTemplate;
+  unsigned char* m_pTemplate = nullptr;
```
Comment (ksooo) [inline tools/depends/native/TexturePacker/src/decoder/GifHelper.h]: Shouldn't we also init `m_gifFile` with a nullptr here (line 113)?

File: xbmc/AppParamParser.h [hunk 3]
```diff
@@ -25,7 +27,7 @@ class CAppParamParser
 
   const CFileItemList& GetPlaylist() const;
 
-  int m_logLevel;
+  int m_logLevel = LOG_LEVEL_NORMAL;
```
Comment (ksooo) [inline xbmc/AppParamParser.h]: This default init required the new include of `commons/ilog.h`, to get the definition of `LOG_LEVEL_NORMAL`, right?

I would always prefer less include files in header files over default  member init to keep compile time dependencies as small as possible...

Here, I would default init with 0 and set the value `LOG_LEVEL_NORMAL` in the cpp file.

File: xbmc/cores/AudioEngine/Utils/AERingBuffer.h [hunk 4]
```diff
@@ -285,11 +273,11 @@ class AERingBuffer {
...
-  unsigned int m_iRead;
-  unsigned int m_iWritten;
-  unsigned int m_iSize;
-  unsigned int m_planes;
+  unsigned int m_iReadPos = 0;
+  unsigned int m_iWritePos = 0;
+  unsigned int m_iRead = 0;
+  unsigned int m_iWritten = 0;
+  unsigned int m_iSize = 0;
+  unsigned int m_planes = 0;
```
Comment (ksooo) [inline xbmc/cores/AudioEngine/Utils/AERingBuffer.h]: Shouldn't we also default init `m_Buffer` with `nullptr` here? (line 282)

File: xbmc/input/actions/Action.h [hunk 5]
```diff
@@ -91,7 +93,7 @@ class CAction
   bool IsAnalog() const;
 
 private:
-  int          m_id;
+  int m_id = ACTION_NONE;
```
Comment (ksooo) [inline xbmc/input/actions/Action.h]: Does default init justify the additional include needed now? IMO not.

File: tools/depends/native/TexturePacker/src/decoder/GifHelper.cpp (hunk 1 above)
Comment (Rechi) [reply tools/depends/native/TexturePacker/src/decoder/GifHelper.cpp]: I know, but it was there before, removed it now manually.

File: tools/depends/native/TexturePacker/src/decoder/GifHelper.h (hunk 2 above)
Comment (Rechi) [reply tools/depends/native/TexturePacker/src/decoder/GifHelper.h]: Have a look at the `GifHelper` constructor.
https://github.com/xbmc/xbmc/blob/dd178e70a87226656ecbc286ff016f419ad4fce9/tools/depends/native/TexturePacker/src/decoder/GifHelper.cpp#L46-L56

File: xbmc/AppParamParser.h (hunk 3 above)
Comment (Rechi) [reply xbmc/AppParamParser.h]: removed this change

File: xbmc/input/actions/Action.h (hunk 5 above)
Comment (Rechi) [reply xbmc/input/actions/Action.h]: removed this change

File: xbmc/cores/AudioEngine/Utils/AERingBuffer.h (hunk 4 above)
Comment (Rechi) [reply xbmc/cores/AudioEngine/Utils/AERingBuffer.h]: fixed manually
`modernize-use-default-member-init` check didn't convert `NULL` and either I didn't run clang-tidy-diff with `modernize-use-nullptr` check on darwin or it didn't work.

**Review by ksooo — APPROVED** [review]
lgtm, thank you

**Pull request merged**